test_parent_retains_lock_across_fork runs into DeprecationWarning on 3.15

Hi, we are seeing the following `DeprecationWarning` with 3.15.0b3 0n 3.29.5.  Not super urgent, 3.15.0 final is planned for [October 1st](https://peps.python.org/pep-0790/).

```python
_____________________ test_parent_retains_lock_across_fork _____________________

tmp_path = PosixPath('/build/pytest-of-nixbld/pytest-0/test_parent_retains_lock_acros0')

    @requires_fork
    @pytest.mark.timeout(15)
    def test_parent_retains_lock_across_fork(tmp_path: Path) -> None:
        path = str(tmp_path / "foo.lock")
        lock = SoftReadWriteLock(path, heartbeat_interval=0.2, stale_threshold=1.0, poll_interval=0.02)
        lock.acquire_write(timeout=5)
        try:
            child = _fork_process(target=time.sleep, args=(0.05,))
>           child.start()

tests/soft_rw/test_soft_rw_sync.py:1051: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/9xszc7da8644vsyfl4gyr4asj3bsx2n9-python3-3.15.0b3/lib/python3.15/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
                  ^^^^^^^^^^^^^^^^^
/nix/store/9xszc7da8644vsyfl4gyr4asj3bsx2n9-python3-3.15.0b3/lib/python3.15/multiprocessing/context.py:290: in _Popen
    return Popen(process_obj)
           ^^^^^^^^^^^^^^^^^^
/nix/store/9xszc7da8644vsyfl4gyr4asj3bsx2n9-python3-3.15.0b3/lib/python3.15/multiprocessing/popen_fork.py:20: in __init__
    self._launch(process_obj)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <multiprocessing.popen_fork.Popen object at 0x7ffff4d98ad0>
process_obj = <ForkProcess name='ForkProcess-20' parent=244 initial>

    def _launch(self, process_obj):
        code = 1
        parent_r, child_w = os.pipe()
        child_r, parent_w = os.pipe()
        # gh-146313: Tell the resource tracker's at-fork handler to keep
        # the inherited pipe fd so this child reuses the parent's tracker
        # (gh-80849) rather than closing it and launching its own.
        from .resource_tracker import _fork_intent
        _fork_intent.preserve_fd = True
        try:
>           self.pid = os.fork()
                       ^^^^^^^^^
E           DeprecationWarning: This process (pid=244) is multi-threaded, use of fork() may lead to deadlocks in the child.

/nix/store/9xszc7da8644vsyfl4gyr4asj3bsx2n9-python3-3.15.0b3/lib/python3.15/multiprocessing/popen_fork.py:76: DeprecationWarning
```

--- a/tests/soft_rw/test_soft_rw_sync.py
+++ b/tests/soft_rw/test_soft_rw_sync.py
@@ -1042,6 +1042,10 @@ def test_child_can_acquire_a_different_lock_after_fork(tmp_path: Path) -> None:
 
 @requires_fork
 @pytest.mark.timeout(15)
+# Holding the write lock keeps the heartbeat thread alive, so this fork is necessarily from a multi-threaded
+# process and Python 3.15 warns that it may deadlock. That is the scenario under test, and it is already safe:
+# register_at_fork resets inherited state in the child (any child use raises "invalidated by fork()"). Expected.
+@pytest.mark.filterwarnings("ignore:.*multi-threaded, use of fork.*:DeprecationWarning")
 def test_parent_retains_lock_across_fork(tmp_path: Path) -> None:
     path = str(tmp_path / "foo.lock")
     lock = SoftReadWriteLock(path, heartbeat_interval=0.2, stale_threshold=1.0, poll_interval=0.02)

